eng/common/pipelines/templates/steps/save-package-properties.yml

```diff
@@ -66,5 +69,5 @@ steps:
           arguments: >
             -ServiceDirectory '${{parameters.ServiceDirectory}}'
             -OutDirectory '${{ parameters.PackageInfoDirectory }}'
-            -AddDevVersion:$${{ eq(variables['SetDevVersion'],'true') }}
+            -AddDevVersion:$${{ and(eq(variables['SetDevVersion'],'true'), eq(parameters.BuildVersion, true)) }}
```

Comment:
I think that just looking at the variables used, I wouldn't remember why we did this. also "BuildVersion" isn't a parameter so I'm not exactly sure how this all plays out.

Reply:
So,
- If we're in a PR
  - Generate a PR diff
  - Tag the build for the service names with changes
  - Save Release package info files
- If we're not a PR
  - Save package info files
    - If we've set "SetDevVersion" to true and some parameter == true
      - Save dev packages
    - Else
      - Save Release Packages